### PR TITLE
Implemented C17 Curse cycle

### PR DESCRIPTION
**Pull request by Eleros**
- updated tokenImageSets of Zombie and Gold tokens
- added Curse of Vitality
- added Curse of Verbosity
- added Curse of Disturbance
- added Curse of Opulence
- added Curse of Bounty
- added Curse cycle to C17

### CODE REVIEW

**Comment by Eleros** [comment]
Just noticed that the curses currently grant the controller the benefit twice if he attacks the enchanted player. Will fix that later..

**Comment by Eleros** [comment]
Should be fixed now.

File: Mage.Sets/src/mage/cards/c/CurseOfDisturbance.java [hunk 1]
```diff
@@ -99,11 +99,14 @@ public boolean checkEventType(GameEvent event, Game game) {
...
     public boolean checkTrigger(GameEvent event, Game game) {
         Permanent enchantment = game.getPermanent(this.getSourceId());
+        UUID controller = this.getControllerId();
         if (enchantment != null
                 && enchantment.getAttachedTo() != null
                 && game.getCombat().getPlayerDefenders(game).contains(enchantment.getAttachedTo())) {
-            for (Effect effect: this.getEffects()) {
-                effect.setTargetPointer(new FixedTarget(game.getCombat().getAttackerId()));                    
+            if (game.getCombat().getAttackerId() != controller) {
+                for (Effect effect: this.getEffects()) {
```
Comment (LevelX2) [inline Mage.Sets/src/mage/cards/c/CurseOfDisturbance.java]: There is  a simpler this.getEffects().setTargetPointer() method for this.

File: Mage.Sets/src/mage/cards/c/CurseOfBounty.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+    public UntapAllNonlandsTargetEffect(final UntapAllNonlandsTargetEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(this.getTargetPointer().getFirst(game, source));
+        if (player != null) {
+            for (Permanent nonland: game.getBattlefield().getAllActivePermanents(new FilterNonlandPermanent(), player.getId(), game)) {
```
Comment (LevelX2) [inline Mage.Sets/src/mage/cards/c/CurseOfBounty.java]: A good case to use a static filter StaticFilters.FILTER_PERMANENT_NON_LAND.

File: Mage.Sets/src/mage/cards/c/CurseOfBounty.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == EventType.DECLARED_ATTACKERS;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        Permanent enchantment = game.getPermanent(this.getSourceId());
```
Comment (Eleros) [inline Mage.Sets/src/mage/cards/c/CurseOfBounty.java]: So I've been looking at older curses and none of them seem to use that method. Mind explaining why this is necessary?
Also, I'm not really sure what that method name means, what's LKI stand for? Would I just replace the getPermanent method here or do I need an if clause to check for destruction before resolution and if so, how do you do that?

File: Mage.Sets/src/mage/cards/c/CurseOfDisturbance.java (hunk 1 above)
Comment (Eleros) [inline Mage.Sets/src/mage/cards/c/CurseOfDisturbance.java]: Not sure what you mean here, mind giving an example?

File: Mage.Sets/src/mage/cards/c/CurseOfBounty.java (hunk 2 above)
Comment (Eleros) [inline Mage.Sets/src/mage/cards/c/CurseOfBounty.java]: Sounds right, will do.

File: Mage.Sets/src/mage/cards/c/CurseOfBounty.java (hunk 3 above)
Comment (LevelX2) [inline Mage.Sets/src/mage/cards/c/CurseOfBounty.java]: LKI = Last known information, a term from the MTG rules.
If the permanent is no longer on the battelfield, the information about the permanent can be retrieved from the LKI storage.
And the method game.getPermanentOrLKIBattlefield(UUID) does take the permanent from the LKI if no permanent with the given id is on the battlefield.
If you only use game.getPermanent() and the object left the battelfield the resultion will fail if you check for the permanent object being != null.

Comment (Eleros) [inline Mage.Sets/src/mage/cards/c/CurseOfBounty.java]: I see. Thanks for the clarification.
Will update that then.

File: Mage.Sets/src/mage/cards/c/CurseOfDisturbance.java [hunk 4]
```diff
@@ -98,12 +98,12 @@ public boolean checkEventType(GameEvent event, Game game) {
...
     public boolean checkTrigger(GameEvent event, Game game) {
-        Permanent enchantment = game.getPermanent(this.getSourceId());
+        Permanent enchantment = game.getPermanentOrLKIBattlefield(this.getSourceId());
         UUID controller = this.getControllerId();
         if (enchantment != null
                 && enchantment.getAttachedTo() != null
                 && game.getCombat().getPlayerDefenders(game).contains(enchantment.getAttachedTo())) {
-            if (game.getCombat().getAttackerId() != controller) {
+            if (!game.getCombat().getAttackerId().equals(controller)) {
                 for (Effect effect: this.getEffects()) {
```
Comment (LevelX2) [inline Mage.Sets/src/mage/cards/c/CurseOfDisturbance.java]: You can use  
this.getEffects().setTargetPointer(new FixedTarget(game.getCombat().getAttackerId()));

**Comment by LevelX2** [comment]
Thanks for your contribution.

**Pull request merged**